povm_toolbox/sampler/povm_sampler_job.py

```diff
@@ -88,6 +90,84 @@ def result(self) -> PrimitiveResult[POVMPubResult]:
 
         return res
 
+    def save_metadata(self, filename: str | None = None) -> None:
+        """Save the metadata of the :class:`.POVMSamplerJob` instance into a JSON file.
+
+        Args:
+            filename: name of the file where to store the metadata.
+        """
+        if filename is None:
+            filename = f"job_metadata_{self.base_job.job_id()}.pkl"
+        with open(filename, "wb") as file:
+            pickle.dump(
+                {
+                    "base_job_id": self.base_job.job_id(),
+                    "metadata": self.metadata,
+                },
```

Comment:
Is there a reason we cannot pickle `self`?

Reply:
I tried but I cannot pickle the internal `BasePrimitiveJob`, this was the error message:

> TypeError: cannot pickle '_thread.RLock' object

Reply:
Okay that's fine :+1: Just wanted to check :slightly_smiling_face: